src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -41,13 +48,20 @@ public EditCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_FIELD);
 
```

Comment:
What if we do 
Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());

Then we perform searchByName(name), to get the Person object. Then we check the exact Class of this Person.

By getting the Class of this Person, for eg : Staff, then we will return 
EditStaffCommandParser(args) to trigger edit staff action

Reply:
ill try it out thanks!